src/Raven.Server/Documents/Handlers/Admin/AdminIndexHandler.cs

```diff
@@ -138,9 +139,13 @@ public async Task OptimizeIndex()
             {
                 var name = GetQueryStringValueAndAssertIfSingleAndNotEmpty("name");
                 var index = Database.IndexStore.GetIndex(name);
+                
                 if (index == null)
                     IndexDoesNotExistException.ThrowFor(name);
 
+                if (index.SearchEngineType == SearchEngineType.Corax)
+                    throw new NotSupportedInCoraxException("Index optimization is not supported in Corax.");
```

Comment:
This is not a place to handle that. I checked the code and it's handled already:

https://github.com/ravendb/ravendb/blob/39c5635f0e8fcfffeed913a7929fd893231b9fa1/src/Raven.Server/Documents/Indexes/Index.cs#L4969-L4972

Reply:
Changed